File: go-course/read_big_file.go
package main

import (
	"fmt"
	"os"
	"time"
)

func main() {
	start := time.Now()

	content, err := os.ReadFile("numbers.txt")
	if err != nil {
		fmt.Println(err)
		return
	}

	fmt.Println(string(content))

	duration := time.Since(start)
	fmt.Printf("Performance: %v\n", duration)
}

File: go-course/read_big_file.js
const fs = require("node:fs")

console.time("performance")

const readStream = fs.createReadStream("numbers.txt")

readStream.on("data", chunk => {
    console.log(chunk.toString())
})

readStream.on("end", () => {
    console.timeEnd("performance")
})

// const content = fs.readFileSync("numbers.txt")
// console.log(content.toString())

// console.timeEnd("performance")